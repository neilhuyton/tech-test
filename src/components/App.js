import React from "react";
import ReportsPage from "./ReportsPage";

const App = () => (
  <div>
    <ReportsPage />
  </div>
);

export default App;
