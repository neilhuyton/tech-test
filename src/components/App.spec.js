import React from "react";
import { configure, mount } from "enzyme";
import { Provider } from "react-redux";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";
import setupStore from "../setupStore";
import toJSON from "enzyme-to-json";
import * as reports from "../__fixtures__/reports.json";
import wait from "waait";

configure({ adapter: new Adapter() });

describe("App", () => {
  let store;

  beforeEach(() => {
    // fetch.resetMocks();
    store = setupStore({ reports: reports.data });
  });

  describe("sorting", () => {
    test("orders reports by ASC by default", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );

      wrapper.update();
      const report = wrapper.find(".report");
      expect(report.length).toEqual(4);
      expect(toJSON(report)).toMatchSnapshot();
    });
    test("correctly orders reports by name when DESC button is clicked", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      wrapper.find(".sort-desc").simulate("click");
      const report = wrapper.find(".report");
      expect(report.length).toEqual(4);
      expect(toJSON(report)).toMatchSnapshot();
    });
    test("correctly orders reports by name when ASC button is clicked", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      wrapper.find(".sort-asc").simulate("click");
      const report = wrapper.find(".report");
      expect(report.length).toEqual(4);
      expect(toJSON(report)).toMatchSnapshot();
    });
  });
  describe("filter", () => {
    test("by age range", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      wrapper.find(".filter-ageRange").simulate("click");
      const report = wrapper.find(".report");
      expect(report.length).toEqual(1);
      expect(toJSON(report)).toMatchSnapshot();
    });
    test("by gender", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      wrapper.find(".filter-gender").simulate("click");
      const report = wrapper.find(".report");
      expect(report.length).toEqual(1);
      expect(toJSON(report)).toMatchSnapshot();
    });

    test("by visitors", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      wrapper.find(".filter-visitors").simulate("click");
      const report = wrapper.find(".report");
      expect(report.length).toEqual(2);
      expect(toJSON(report)).toMatchSnapshot();
    });
  });
  describe("deleting", () => {
    test("a report, removes it from the UI", async () => {
      let wrapper = mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
      wrapper.update();

      let report = wrapper.find(".report");
      expect(report.length).toEqual(4);

      wrapper
        .find(".report-delete")
        .first()
        .simulate("click");

      await wait(2000);

      wrapper.update();
      report = wrapper.find(".report");

      expect(report.length).toEqual(3);
      expect(toJSON(report)).toMatchSnapshot();
    });
  });
});
