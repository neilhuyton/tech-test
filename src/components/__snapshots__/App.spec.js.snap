// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App deleting a report, removes it from the UI 1`] = `
Array [
  <tr
    className="report"
  >
    <td>
      Thirty day gender report
    </td>
    <td>
      Gender
    </td>
    <td>
      pie
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day visitor report
    </td>
    <td>
      Visitors
    </td>
    <td>
      bar
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Weekly age range report
    </td>
    <td>
      Age range
    </td>
    <td>
      bar
    </td>
    <td>
      weekly
    </td>
    <td>
      false
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
]
`;

exports[`App filter by age range 1`] = `
<tr
  className="report"
>
  <td>
    Weekly age range report
  </td>
  <td>
    Age range
  </td>
  <td>
    bar
  </td>
  <td>
    weekly
  </td>
  <td>
    false
  </td>
  <td>
    <button
      className="report-delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;

exports[`App filter by gender 1`] = `
<tr
  className="report"
>
  <td>
    Thirty day gender report
  </td>
  <td>
    Gender
  </td>
  <td>
    pie
  </td>
  <td>
    monthly
  </td>
  <td>
    true
  </td>
  <td>
    <button
      className="report-delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;

exports[`App filter by visitors 1`] = `
Array [
  <tr
    className="report"
  >
    <td>
      Daily visitors report
    </td>
    <td>
      Visitors
    </td>
    <td>
      column
    </td>
    <td>
      daily
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day visitor report
    </td>
    <td>
      Visitors
    </td>
    <td>
      bar
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
]
`;

exports[`App sorting correctly orders reports by name when ASC button is clicked 1`] = `
Array [
  <tr
    className="report"
  >
    <td>
      Daily visitors report
    </td>
    <td>
      Visitors
    </td>
    <td>
      column
    </td>
    <td>
      daily
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day gender report
    </td>
    <td>
      Gender
    </td>
    <td>
      pie
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day visitor report
    </td>
    <td>
      Visitors
    </td>
    <td>
      bar
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Weekly age range report
    </td>
    <td>
      Age range
    </td>
    <td>
      bar
    </td>
    <td>
      weekly
    </td>
    <td>
      false
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
]
`;

exports[`App sorting correctly orders reports by name when DESC button is clicked 1`] = `
Array [
  <tr
    className="report"
  >
    <td>
      Weekly age range report
    </td>
    <td>
      Age range
    </td>
    <td>
      bar
    </td>
    <td>
      weekly
    </td>
    <td>
      false
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day visitor report
    </td>
    <td>
      Visitors
    </td>
    <td>
      bar
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day gender report
    </td>
    <td>
      Gender
    </td>
    <td>
      pie
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Daily visitors report
    </td>
    <td>
      Visitors
    </td>
    <td>
      column
    </td>
    <td>
      daily
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
]
`;

exports[`App sorting orders reports by ASC by default 1`] = `
Array [
  <tr
    className="report"
  >
    <td>
      Daily visitors report
    </td>
    <td>
      Visitors
    </td>
    <td>
      column
    </td>
    <td>
      daily
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day gender report
    </td>
    <td>
      Gender
    </td>
    <td>
      pie
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Thirty day visitor report
    </td>
    <td>
      Visitors
    </td>
    <td>
      bar
    </td>
    <td>
      monthly
    </td>
    <td>
      true
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
  <tr
    className="report"
  >
    <td>
      Weekly age range report
    </td>
    <td>
      Age range
    </td>
    <td>
      bar
    </td>
    <td>
      weekly
    </td>
    <td>
      false
    </td>
    <td>
      <button
        className="report-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>,
]
`;
